src/DynamoCoreWpf/ViewModels/Search/BrowserItemViewModel.cs

```diff
@@ -470,16 +483,22 @@ private ImageSource LoadDefaultIcon()
 
         public void InsertSubCategory(NodeCategoryViewModel newSubCategory)
         {
-            var first = SubCategories.Select((x, i) => new { x.Name, Idx = i })
-                .FirstOrDefault(
-                    x =>
-                        string.Compare(
-                            x.Name,
-                            newSubCategory.Name,
-                            StringComparison.Ordinal)
-                            >= 0);
-            if (first != null)
-                SubCategories.Insert(first.Idx, newSubCategory);
+            bool hasClasses = SubCategories.FirstOrDefault() is ClassesNodeCategoryViewModel;
+            var nextLargerItemIndex = -1;
+            foreach (var subCategory in SubCategories.Where(cat => !(cat is ClassesNodeCategoryViewModel)))
+            {
+                if (string.Compare(subCategory.Name, newSubCategory.Name, StringComparison.Ordinal) >= 0)
+                {
+                    nextLargerItemIndex = Items.IndexOf(subCategory);
+                    break;
+                }
+            }
```

Comment:
This loop seems to be new, in that case I think we can fold the two loops (this and the one above) into a helper method:

```csharp
internal static int FindInsertionPointByName(
    IEnumerable<ISearchEntryViewModel> list, string name)
{
}
```

Then the callers can do something like this:

```csharp
var list = SubCategories.Where(cat => !(cat is ClassesNodeCategoryViewModel));
var nextLargerItemIndex = FindInsertionPointByName(list, newSubCategory.Name);
```

Then you can write a test case for that with some dummy data to ensure `-1` and valid indices are returned.